traits/trait_dict_object.py

```diff
@@ -195,6 +196,39 @@ def __delitem__(self, key):
         super().__delitem__(key)
         self.notify(removed=removed, added={}, changed={})
 
+    if sys.version_info >= (3, 9):
+        def __ior__(self, other):
+            """ Update self with the contents of other.
+
+            Parameters
+            ----------
+            other : mapping or iterable of (key, value) pairs
+                Values to be added to this dictionary.
+            """
+            validated_dict = {}
+            added = {}
+            changed = {}
+
+            items = other.items() if hasattr(other, 'keys') else other
+
+            for key, value in items:
+                validated_key = self.key_validator(key)
+                validated_value = self.value_validator(value)
+
+                if validated_key in self:
+                    changed[validated_key] = self[validated_key]
+                else:
+                    added[validated_key] = validated_value
+
+                validated_dict[validated_key] = validated_value
+
+            retval = super().__ior__(validated_dict)
+
+            if added or changed:
```

Comment:
There is a difference in behaviour here between this and `update`:
```
x.a |= {}
```
will not notify, but
```
x.a.update({})
```
does.

I would expect them to be consistent, and this may lead to issues in the future if people replace `.update` with `|=` and expect the same behaviour.